modules/3d/include/opencv2/3d/ptcloud.hpp

```diff
@@ -11,6 +11,153 @@ namespace cv {
 //! @addtogroup _3d
 //! @{
 
+
+//! type of the robust estimation algorithm
+enum SacMethod
+{
+    /** The RANSAC algorithm described in @cite fischler1981random.
+     */
+    SAC_METHOD_RANSAC,
+    //    SAC_METHOD_MAGSAC,
+    //    SAC_METHOD_LMEDS,
+    //    SAC_METHOD_MSAC,
+    //    SAC_METHOD_RRANSAC,
+    //    SAC_METHOD_RMSAC,
+    //    SAC_METHOD_MLESAC,
+    //    SAC_METHOD_PROSAC
+};
+
+enum SacModelType
+{
+    /** The 3D PLANE model coefficients in list **[a, b, c, d]**,
+     corresponding to the coefficients of equation
+     \f$ ax + by + cz + d = 0 \f$. */
+    SAC_MODEL_PLANE,
+    /** The 3D SPHERE model coefficients in list **[center_x, center_y, center_z, radius]**,
+     corresponding to the coefficients of equation
+     \f$ (x - center\_x)^2 + (y - center\_y)^2 + (z - center\_z)^2 = radius^2 \f$.*/
+    SAC_MODEL_SPHERE,
+    //    SAC_MODEL_CYLINDER,
+
+};
+
+
+/** @brief Sample Consensus algorithm segmentation of 3D point cloud model.
+
+Example of segmenting plane from a 3D point cloud using the RANSAC algorithm:
+@snippet snippets/3d_sac_segmentation.cpp planeSegmentationUsingRANSAC
+
+@see
+1. Supported algorithms: enum SacMethod in ptcloud.hpp.
+2. Supported models: enum SacModelType in ptcloud.hpp.
+ */
+class CV_EXPORTS SACSegmentation
+{
+public:
+    /** @brief Custom function that take the model coefficients and return whether the model is acceptable or not.
+
+     Example of constructing SACSegmentation::ModelConstraintFunction:
+     @snippet snippets/3d_sac_segmentation.cpp usageExampleSacModelConstraintFunction
+
+     @note The content of model_coefficients depends on the model.
+     Refer to the comments inside enumeration type SacModelType.
+     */
+    using ModelConstraintFunction =
+    std::function<bool(const std::vector<double> &/*model_coefficients*/)>;
+
+    //-------------------------- CREATE -----------------------
+
+    static Ptr<SACSegmentation> create(SacModelType sac_model_type_ = SAC_MODEL_PLANE,
+            SacMethod sac_method_ = SAC_METHOD_RANSAC,
+            double threshold_ = 0.5, int max_iterations_ = 1000);
```

Comment:
> sac_model_type_

Please avoid using underscores suffixes/prefixes in public API declarations, especially for parameters (they become part of documentation and bindings).

Reply:
fixed.